scripts/rotate-swiss-vault.sh

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env bash
+
+set -e  # Exit on errors
+
+init_keepass () {
+    # Locate keepassxc-cli or fail with error
+    if [[ -x "$KEE_PASS" ]]; then
+        echo "Running with explicit KeePass binary at: $KEE_PASS"
+    elif [[ "$OSTYPE" == "darwin"* ]]; then
+        KEE_PASS=/Applications/KeePassXC.app/Contents/MacOS/keepassxc-cli
+    elif [[ "$OSTYPE" == "linux-gnu"* ]]; then
+        KEE_PASS=`which keepassxc-cli`
+    fi
+
+    if [ ! -x "$KEE_PASS" ]; then
+        echo "KeePassXC was not found. Please go to https://keepassxc.org to install it"
+        echo "or set the KEE_PASS environment variable to the path of keepassxc-cli"
+        exit 1
+    fi
+}
+
+prompt_for_password () {
+    echo "Enter password to unlock $KEE_PASS_DB:"
+    read -s KEE_PASS_DB_PASS
+}
+
+store_password () {
+    local password="$1"
+
+    { echo "$KEE_PASS_DB_PASS"; echo "$KEE_PASS_DB_PASS"; } | \
+        "$KEE_PASS" db-create "$TEMP_DATABASE" -p > /dev/null 2>&1
+    { echo "$KEE_PASS_DB_PASS"; echo "$password"; } | \
+        "$KEE_PASS" add "$TEMP_DATABASE" "$TEMP_KEY" -p -q 
```

Comment:
Does this mean that these passwords _cannot_ have an actual newline in them (these commands would certainly fail if the db password contained a newline).  Maybe guard against that scenario? Could be done here, but might be better to check immediately after each password is first known to the script.

```bash
if [ $(echo "$KEE_PASS_DB_PASS" | wc -l) -ne 1 ]; then
    echo 'invalid db password (passwords may not contain newlines)'
    exit 1
fi

# snip ...

if [ $(echo "$password" | wc -l) -ne 1 ]; then
    echo 'invalid vault password (passwords may not contain newlines)'
    exit 1
fi
```

Reply:
My testing works fine with newlines. I don't think multiple echo command emit a '\n' character

Reply:
> a developer would have to go out of their way to intentionally troll us

Or `keepassxc-cli generate` (or whatever binary the user provides in its place) emits more than one line of text on STDOUT.  Since the value comes from an executable outside of this script's "control", I think it's important to employ the robustness principle here (i.e. this script should refuse to encrypt the vault with value that the user does not know, and it can detect is a malformed password).  If the "password generate" command for some reason wrote multiple lines of output, the end user would have no idea something other than what they expected actually happened.

Consider the following where:
- `KEE_PASS_DB_PASS='something the user typed'`
- `password='multi-line↲output↲'` (Note: `↲` represents a newline (ASCII character 10) -- I know, that's the unicode symbol for a carriage return, but you get the idea).

```bash
    { echo "$KEE_PASS_DB_PASS"; echo "$password"; } | \
        "$KEE_PASS" add "$TEMP_DATABASE" "$TEMP_KEY" -p -q
```
The key used to encrypt the vault is `multi-line↲output`, and the value saved in the temporary database is `multi-line`, but the user has no way of knowing this. In this scenario, all of the following are true:
- the vault is encrypted with a new key
- the user can unlock the keepass db with the password they provided (i.e. the db appears fine)
- the value stored in the keepass db is incorrect (cannot unlock the vault)
- the user does not know that the value in the keepass db is incorrect

Reply:
Another solution could be (if we're willing to assume the _first_ line of output is always a "good password".
```bash
GENERATED_PASS=`"$KEE_PASS" generate | head -n1`
```

Reply:
I still feel like this sounds like over-engineering for a situation that will never happen, and should never happen. I can't find any option within KeePass to generate a password with a newline (apart from perhaps manually adding that character in the 'Additional Characters' field). I configured 1Pass to look at its API, and it provides even less configurability with passwords than KeePass does. A password with a newline character is going to cause issues all over the place -- it's not a coincidence that these generators don't give you the option to include that character.

Sure, I could add this check to make it more robust, but if that scenario should never come up, it's additional code that has to be understood and maintained. In javascript, for example, I don't tend to check that my string parameters are actually strings and not numbers, unless they could actually be passed as numbers by mistake. I just don't think there is any realistic case where a password contains a newline.

That said, if my assumptions are incorrect, and this does happen, it is recoverable. The encrypted vault can be rolled back to its previous version in GitHub, and both KeePass and 1Pass have records of their previous passwords, so we can restore the version prior to the faulty one.

Reply:
> it's additional code that has to be understood and maintained
:heavy_plus_sign: 

I'm satisfied to leave it as-is.